### PR TITLE
Extension block rendering

### PR DESCRIPTION
**Pull request by ericrosenbaum**
### Resolves

Resolves https://github.com/LLK/scratch-blocks/issues/1207

Note that this PR does not include the vertical separator line after the icon shown in the spec- I'll make a separate issue to add that part.

### Proposed Changes

Update the blocks rendering to bring blocks with extension icons on them up to spec. This includes:

- Increasing the height of command and reporter blocks
- Adding horizontal offset to the icon in some cases
- Adding vertical offset to the icon in some cases
 
### Reason for Changes

To accommodate the icon on extension blocks at the size we want, we needed to make some layout changes to the blocks.

### Test Coverage

I've added an extension blocks category to the vertical playground via the default toolbox, to visually test out the rendering.

### CODE REVIEW

**Review by rachel-fenichel — COMMENTED** [review]
Before I dive too much into the code:
- This change effectively makes any block with a leading image render as an extension block.  Is that a design that you're comfortable with?  If so, it needs to be communicated to anyone developing with the library.
- Please handle RTL, and test in RTL.

**Comment by ericrosenbaum** [comment]
@rachel-fenichel re: communication, I guess it's more like: any block with a leading image will be rendered in such a way that it is laid out properly according to our spec (there's nothing extension-specific in particular about the rendering- it's just making the blocks sized to fit the images properly). The code I wrote assumes that these leading images are 40x40px. In our spec for designers, we will of course explain this requirement as well.   

I'll have a look at RTL- I forgot to do that, and it's definitely not quite right yet. It's more noticeable once you get to my next PR with the vertical separator lines.

**Comment by rachel-fenichel** [comment]
Right, I just want to be clear that the spec is effectively for all blocks with leading images, rather than being a spec for extension blocks specifically.  Similarly, the next PR makes all blocks that lead with an image have a bar after the image, regardless of whether it's intended to be an extension block.

Thanks for fixing RTL.  I'll take a closer look at the code now.

File: blocks_vertical/extensions.js
Comment (rachel-fenichel) [inline blocks_vertical/extensions.js:62]: The Blockly documentation [says](https://developers.google.com/blockly/guides/create-custom-blocks/define-blocks#image) that you can use a relative path here, so you shouldn't need to include the `Blockly.mainWorkspace.options.pathToMedia` part.

File: blocks_vertical/extensions.js
Comment (rachel-fenichel) [inline blocks_vertical/extensions.js:35]: You can use Blockly.defineBlocksWithJsonArray for this: https://github.com/LLK/scratch-blocks/blob/develop/core/blockly.js#L358

File: core/block_render_svg_vertical.js
Comment (rachel-fenichel) [inline core/block_render_svg_vertical.js]: Indent continuation lines +4 spaces, here and everywhere: https://google.github.io/styleguide/jsguide.html#formatting-indent

File: core/block_render_svg_vertical.js
Comment (rachel-fenichel) [inline core/block_render_svg_vertical.js]: Nit: line wrap at 80

**Review by rachel-fenichel — COMMENTED** [review]
I have a few style nits, but LGTM after that.

File: blocks_vertical/extensions.js
Comment (ericrosenbaum) [reply blocks_vertical/extensions.js:62]: We're using this pattern across 13 files- that path variable has been set to "../media/", so it's already relative. Is it okay to leave it as is?

File: blocks_vertical/extensions.js
Comment (ericrosenbaum) [reply blocks_vertical/extensions.js:35]: I just followed the pattern of the rest of the blocks definition files in that directory. It looks like defineBlocksWithJsonArray is used mainly in the unit tests. Is it okay if I leave it as is? Or is there some reason that defineBlocksWithJsonArray is better?

File: core/block_render_svg_vertical.js
Comment (ericrosenbaum) [reply core/block_render_svg_vertical.js]: all set

File: core/block_render_svg_vertical.js
Comment (ericrosenbaum) [reply core/block_render_svg_vertical.js]: all set

File: blocks_vertical/extensions.js
Comment (rachel-fenichel) [reply blocks_vertical/extensions.js:35]: Discussed offline; doesn't need to happen in this PR, but should happen eventually.

File: blocks_vertical/extensions.js
Comment (rachel-fenichel) [reply blocks_vertical/extensions.js:62]: There's a code smell here, but not one that we need to handle now.  Leave as is.

**Comment by rachel-fenichel** [comment]
After offline discussion: this is fine for now, and as a followup Eric will use an `isScratchExtension` flag to check if it's an extension block, instead of checking whether it starts with an image.

**Review by rachel-fenichel — APPROVED** [review]

**Pull request merged**
